### ISSUE TITLE
different dimension reduction methods  and feature importance scores

### ISSUE THREAD

**Issue by wangqianwen0418** [hms-dbmi/OncoThreads#258]
**dimension reduction methods for the scatter plot**:
- [x] pca
- [x] umap
- [x] tsne

**importance score**
- [ ] mutual information

    References
    .. [1] A. Kraskov, H. Stogbauer and P. Grassberger, "Estimating mutual
           information". Phys. Rev. E 69, 2004.
    .. [2] B. C. Ross "Mutual Information between Discrete and Continuous
           Data Sets". PLoS ONE 9(2), 2014.

**Comment by wangqianwen0418** [comment]
**TSNE:**
 dim: 2,
perplexity: 10,
                            earlyExaggeration: 4.0,
                            learningRate: 100.0,
                            nIter: 500,
                            metric: 'euclidean'
<img src="https://user-images.githubusercontent.com/19774198/104391614-40ccd180-550e-11eb-9f94-8d9ca49b87c2.png" width="300"/>

**UMAP:**
nComponents: 2,
nEpochs: 400,
                            nNeighbors: 15

<img src="https://user-images.githubusercontent.com/19774198/104392474-1b40c780-5510-11eb-995e-9adb8bcb7f1c.png" width="300"/>

**PCA:**
<img src="https://user-images.githubusercontent.com/19774198/104392726-b2a61a80-5510-11eb-89a1-a3f5254def69.png" width="300"/>

**To sum up:**
- in terms of performance, UMAP is better than TSNE.
- TSNE and UMAP both focus on global structure while PCA also reserves local structure  
- in terms of Speed, PCA > UMAP > TSNE

**Comment by wangqianwen0418** [comment]
**importance score**
- previously, I simply use the eigenvalue value in pca as the importance score of each feature. Eigenvalue indicates how importance this feature is for the 2d embedding. However, the feature importance in the 2d embedding is not necessarily the feature importance in the state identification.  
Below is an example:

covid 19:

- UMAP:
![image](https://user-images.githubusercontent.com/19774198/104622477-500f6480-565f-11eb-8650-c98d28f0575e.png)

- PCA:
 ![image](https://user-images.githubusercontent.com/19774198/104494019-e2eac900-55a3-11eb-85cd-64bab1de1dba.png)


- I am trying to use mutual information between one feature and the state identification results to represent the importance score of the feature.

**Comment by wangqianwen0418** [comment]
support customization of the DR method.
<img src="https://user-images.githubusercontent.com/19774198/104627588-36711b80-5665-11eb-9332-358f859c7359.png" width="400"/>
